src/main/java/no/unit/nva/model/contexttypes/Book.java

```diff
@@ -21,70 +19,61 @@
 public class Book implements BasicContext {
 
     public static final ISBNValidator ISBN_VALIDATOR = new ISBNValidator();
-    public static final Pattern EXPECTED_SERIES_URI_PATTERN =
-        Pattern.compile("https://.*?nva\\.aws\\.unit\\.no/publication-channels/.*");
-    @Deprecated
-    private String seriesTitle;
-    private URI seriesUri;
-    private String seriesNumber;
-    private String publisher;
-    private List<String> isbnList;
-
-    public Book() {
-    }
 
-    @JacocoGenerated
-    @Override
-    public int hashCode() {
-        return Objects.hash(getSeriesTitle(), getSeriesUri(), getSeriesNumber(), getPublisher(), getIsbnList());
+    private final BookSeries series;
+    private final String seriesNumber;
+    private final String publisher;
+    private final List<String> isbnList;
+
+    public Book(@JsonProperty("series") BookSeries series,
+                @JsonProperty("seriesTitle") String seriesTitle,
+                @JsonProperty("seriesNumber") String seriesNumber,
+                @JsonProperty("publisher") String publisher,
+                @JsonProperty("isbnList") List<String> isbnList) throws InvalidIsbnException {
+        if (nonNull(seriesTitle) && isNull(series)) {
+            this.series = new UnconfirmedSeries(seriesTitle);
+        } else {
+            this.series = series;
```

Comment:
I suggest not allowing both series field and seriesTitle to be set. 
suggestion:
```
public Book(@JsonProperty("series") BookSeries series,
                @JsonProperty("seriesTitle") String seriesTitle,
                @JsonProperty("seriesNumber") String seriesNumber,
                @JsonProperty("publisher") String publisher,
                @JsonProperty("isbnList") List<String> isbnList) throws InvalidIsbnException {
        this.series = setUpSeriesEntry(series, seriesTitle);
...
}

    private BookSeries setUpSeriesEntry(BookSeries series, String seriesTitle) {

        if (isUnconfirmedSeries(series, seriesTitle)) {
            return new UnconfirmedSeries(seriesTitle);
        }
        if (isConfirmedSeries(series)) {
            return series;
        }
        throw new IllegalArgumentException("Ambiguous series definition, both unconfirmed series title and confirmed "
                                           + "series object present");
    }

    private boolean isConfirmedSeries(BookSeries series) {
        return nonNull(series);
    }

    private boolean isUnconfirmedSeries(BookSeries series, String seriesTitle) {
        return nonNull(seriesTitle) && isNull(series);
    }
```

Reply:
In a way, I disagree — this breaks the Robustness principle. As I have coded it, I accept a seriesTitle string and seriesObject without side-effects. Admittedly, if you submit two different values, then we have a problem.